LibDeflate.lua

```diff
@@ -3332,19 +3560,54 @@ LibDeflate.internals = {
 \-d    do decompression instead of compression.
 \--dict <filename> specify the file that contains
 the entire preset dictionary.
+\--gzip  use gzip format instead of raw deflate.
 \-h    give this help.
 \--strategy <fixed/huffman_only/dynamic> specify a special compression strategy.
 \-v    print the version and copyright info.
 \--zlib  use zlib format instead of raw deflate.
 ]]
 
+-- support ComputerCraft shell
+local arg = _G.arg
+local debug = debug
+if shell then 
+    arg = {...} 
```

Comment:
Is "shell" a global variable? If yes, use "_G.shell" to access it.
I prefer to use another way to check if ComputerCraft shell exists.
Check _G.shell seems not safe.

Reply:
Shell is an environmental variable, so I can't use _G.shell to check it.

Reply:
Also, shell is only available when called from the shell.

Reply:
The more standard way to fetch environment variable is to use "os.getenv" method.
Does ComputerCraft create "shell" variable as a local variable?
If that is the case, use "debug.getlocal" to get the value of "shell" variable.

Don't access non-standard global variable like this. Also,

Reply:
ComputerCraft has neither os.getenv or the debug library, but it does have an _ENV upvalue as in Lua 5.2+, as well as getfenv() as in Lua 5.1. Neither of these work in all Lua versions, though, so I'm not sure which one I should check for, since both ways will break for at least one version. (`_ENV.shell == getfenv(1).shell` = true)

Reply:
Plan to remove direct support for ComputerCraft specific APIs in the code of LibDeflate.